doc/developers/develop.rst

```diff
@@ -659,15 +659,6 @@ Even if it is not recommended, it is possible to override the method
 any of the keys documented above is not present in the output of `_get_tags()`,
 an error will occur.
 
-In addition to the tags, estimators also need to declare any non-optional
```

Comment:
This is now not required, and there's no need for a replacement since we now pass instances to estimator checks and not classes.